src/redux/reducer.js

```diff
@@ -7,13 +8,38 @@ import {
   SET_LOADING,
 } from './actions';
 
+const getStoryTimeEntries = timeEntries => {
+  return timeEntries.filter(time => time.reference).map(time => {
+    return {
+      hours: time.hours,
+      reference: parseInt(time.reference, 10),
+    };
+  });
+};
+
+const totalStoryHours = (story, timeEntries) =>
+  timeEntries.reduce(
+    (sum, time) => (time.reference === story.number ? sum + time.hours : sum),
+    0,
+  );
+
 function reduceReceiveEntities(state, action) {
+  const entities = {
+    ...state.entities,
+    ...action.payload,
+  };
+
+  if (!isEmpty(entities.timeEntries) && !isEmpty(entities.stories)) {
```

Comment:
The typical best practice in a redux app like this is to "not store derived state" in the store. In this case, since we can derive the `totalHours` from the state that's already in there (`timeEntries` + `stories`). The refactor to avoid the derived data in the store is to shift the derivation into a selector.

In the current factoring of the app, there's some confusion of concepts. We have a "service" in `analyzeBurndown` that's really a selector deriving data from state. So we're not perfect on this front as it stands already.

It looks like the place this code "should go" is inside the [`reportSelector`](https://github.com/b2io/kubera/commit/fb62324a9fa61c2e97978147078ef35e8906d431). Hopefully that makes sense, let me know if you have any questions about how to structure that.

Past just moving the code there, you could also consider using [`reselect`](https://github.com/reactjs/reselect) to memoize the calculation (since it's probably a bit expensive). I can talk more about how that's done whenever you have some time to chat.

Reply:
It doesn't seem like the calculation is happening more than once, but I could be wrong. I'm interested in discussing the potential problem and how `reselect` can solve it.